packages/flutter_tools/lib/src/project.dart

```diff
@@ -66,75 +84,52 @@ class FlutterProject {
-          .childFile('Generated.xcconfig');
-    });
-  }
-  Future<File> _generatedXcodeProperties;
 
-  File get flutterPluginsFile {
-    return _flutterPluginsFile ??= directory.childFile('.flutter-plugins');
+  File get generatedXcodePropertiesFile {
+    return directory
+        .childDirectory(manifest.isModule ? '.ios' : 'ios')
+        .childDirectory('Flutter')
+        .childFile('Generated.xcconfig');
   }
-  File _flutterPluginsFile;
-
-  Future<Directory> get androidPluginRegistrantHost async {
-    return _androidPluginRegistrantHost ??= manifest.then((FlutterManifest manifest) {
-      if (manifest.isModule) {
-        return directory.childDirectory('.android').childDirectory('Flutter');
-      } else {
-        return directory.childDirectory('android').childDirectory('app');
-      }
-    });
+
+  File get flutterPluginsFile => directory.childFile('.flutter-plugins');
+
+  Directory get androidPluginRegistrantHost {
+    return manifest.isModule
+        ? directory.childDirectory('.android').childDirectory('Flutter')
+        : directory.childDirectory('android').childDirectory('app');
   }
-  Future<Directory> _androidPluginRegistrantHost;
-
-  Future<Directory> get iosPluginRegistrantHost async {
-    return _iosPluginRegistrantHost ??= manifest.then((FlutterManifest manifest) {
-      if (manifest.isModule) {
-        // In a module create the GeneratedPluginRegistrant as a pod to be included
-        // from a hosting app.
-        return directory
-            .childDirectory('.ios')
-            .childDirectory('Flutter')
-            .childDirectory('FlutterPluginRegistrant');
-      } else {
-        // For a non-module create the GeneratedPluginRegistrant as source files
-        // directly in the iOS project.
-        return directory.childDirectory('ios').childDirectory('Runner');
-      }
-    });
+
+  Directory get iosPluginRegistrantHost {
+    // In a module create the GeneratedPluginRegistrant as a pod to be included
+    // from a hosting app.
+    // For a non-module create the GeneratedPluginRegistrant as source files
+    // directly in the iOS project.
+    return manifest.isModule
+        ? directory.childDirectory('.ios').childDirectory('Flutter').childDirectory('FlutterPluginRegistrant')
+        : directory.childDirectory('ios').childDirectory('Runner');
   }
-  Future<Directory> _iosPluginRegistrantHost;
 
-  /// Returns true if this project has an example application
-  bool get hasExampleApp => _exampleDirectory.childFile('pubspec.yaml').existsSync();
+  /// The example sub-project of this project.
+  FlutterProject get example => new FlutterProject._(_exampleDirectory, _exampleManifest, FlutterManifest.empty());
 
-  /// The example sub project of this (package or plugin) project.
-  FlutterProject get example => new FlutterProject(_exampleDirectory);
+  /// True, if this project has an example application
+  bool get hasExampleApp => _exampleDirectory.childFile('pubspec.yaml').existsSync();
```

Comment:
Should this be done at construction time rather than when called (since it's supposed to be an immutable snapshot?). Or maybe checking _exampleManifest?

Reply:
I agree, `hasExampleApp` should not check `example/pubspec.yaml` on-the-fly as it muddles the snapshot semantics. I'll find a better way.

Reply:
Leaving it to another PR, as it "a better way" entails more changes.